### ISSUE TITLE
Unknown type in try/except clause

### ISSUE THREAD

**Issue by d-tw** [microsoft/pyright#1604]
**Describe the bug**
When catching a dynamic Exception type in a try/except clause, the type of the exception instance is `unknown`, when it should be the upper bound of the type variable.

**To Reproduce**
See the screenshot/code.

**Expected behavior**
The `ex` variable on line 10 should be typed to `Exception`, as that is the bound of the `E` type variable.

**Screenshots or Code**
<img width="603" alt="image" src="https://user-images.githubusercontent.com/3317531/110517591-56740700-810b-11eb-9456-bf4ba2085adb.png">

```python
from typing import Callable, Type, TypeVar


E = TypeVar('E', bound=Exception)


def map_exception(exc_type: Type[E], fn: Callable[..., None]) -> None:
    try:
        fn()
    except exc_type as ex:
        ...
```

**VS Code extension or command-line**
pyright 1.1.119

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.120, which I just published. It will also be included in the next release of pylance.
